specs/accounting_reform/beacon-chain.md

```diff
@@ -278,8 +282,43 @@ def process_rewards_and_penalties(state: BeaconState) -> None:
         get_standard_flag_deltas(state, flag, numerator) for (flag, numerator) in FLAGS_AND_NUMERATORS
     ]
     deltas = flag_deltas + [get_inactivity_penalty_deltas]
-    for (rewards, penalties) in deltas:
+    for (rewards, penalty) in deltas:
         for index in range(len(state.validators)):
             increase_balance(state, ValidatorIndex(index), rewards[index])
-            decrease_balance(state, ValidatorIndex(index), penalties[index])
+        state.balance_denominator = state.balance_denominator + penalty + (state.balance_denominator - GWEI_PER_ETH) * penalty // GWEI_PER_ETH
+```
+
+#### `process_final_updates`
+
+```python
+def process_final_updates(state: BeaconState) -> None:
+    current_epoch = get_current_epoch(state)
+    next_epoch = Epoch(current_epoch + 1)
+    # Reset eth1 data votes
+    if next_epoch % EPOCHS_PER_ETH1_VOTING_PERIOD == 0:
+        state.eth1_data_votes = []
+    # Update effective balances with hysteresis
+    for index, validator in enumerate(state.validators):
+        balance_increments = state.balances[index] * HYSTERESIS_QUOTIENT // state.balance_denominator
```

Comment:
Suggest name `balance_in_increments` to make it clear that `increment` is menat as a unit here